### PR TITLE
(FACT-2559) Fix Facter.debugging? call when Facter not fully loaded

### PR DESCRIPTION
**Pull request by Filipovici-Andrei**

### CODE REVIEW

File: lib/framework/logging/logger.rb [hunk 1]
```diff
@@ -94,5 +94,15 @@ def error(msg, colorize = false)
...
     end
+
+    private
+
+    def debugging_active?
+      if Facter.respond_to?(:debugging?)
+        Facter.debugging?
+      else
+        true
+      end
```
Comment (gimmyxd) [inline lib/framework/logging/logger.rb:105]: this can be written as 
```ruby
Facter.respond_to?(:debugging?) && Facter.debugging?
```

File: spec/framework/logging/logger_spec.rb [hunk 2]
```diff
@@ -3,91 +3,123 @@
...
 
   describe '#debug' do
     before do
       allow(Facter).to receive(:debugging?).and_return(true)
     end
 
-    let(:handler) { instance_spy(Logger) }
-
-    it 'noops of debugging is not set' do
+    it 'no debug messages are sent if debugging is set to false' do
```
Comment (gimmyxd) [inline spec/framework/logging/logger_spec.rb:17]: usually `if` statements from example should be a context

```ruby
context 'when debugging is false' do
	it 'does not send a message'
	end
end
``` 

http://www.betterspecs.org/#contexts

File: spec/framework/logging/logger_spec.rb [hunk 3]
```diff
@@ -3,91 +3,123 @@
...
+        expect(multi_logger_double).to have_received(:debug).with('Class - debug_message')
+      end
+    end
+
+    it_behaves_like 'writes debug message'
+
+    context 'when message callback is provided' do
+      after do
+        Facter::Log.class_variable_set(:@@message_callback, nil)
+      end
```
Comment (gimmyxd) [inline spec/framework/logging/logger_spec.rb:50]: Why is this needed? is the logger keeping `messasge_callback` between calls?

File: spec/framework/logging/logger_spec.rb (hunk 2 above)
Comment (Filipovici-Andrei) [reply spec/framework/logging/logger_spec.rb:17]: done

File: lib/framework/logging/logger.rb (hunk 1 above)
Comment (Filipovici-Andrei) [reply lib/framework/logging/logger.rb:105]: done

File: spec/framework/logging/logger_spec.rb (hunk 3 above)
Comment (Filipovici-Andrei) [reply spec/framework/logging/logger_spec.rb:50]: Yes, the logger is keeping the message_callback between calls

File: lib/framework/logging/logger.rb (hunk 1 above)
Comment (Filipovici-Andrei) [reply lib/framework/logging/logger.rb]: yes, you are correct

File: spec/framework/logging/logger_spec.rb (hunk 3 above)
Comment (Filipovici-Andrei) [reply spec/framework/logging/logger_spec.rb]: I'm not sure if this is the expected behaviour

**Review by sebastian-miclea — DISMISSED** [review]

**Review by BogdanIrimie — APPROVED** [review]
